presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -71,12 +75,16 @@
     private final Duration infoCacheTime;
     private final Duration clientTimeout;
 
+    private final LocalMemoryManager localMemoryManager;
     private final LoadingCache<QueryId, QueryContext> queryContexts;
     private final LoadingCache<TaskId, SqlTask> tasks;
 
     private final SqlTaskIoStats cachedStats = new SqlTaskIoStats();
     private final SqlTaskIoStats finishedTaskStats = new SqlTaskIoStats();
 
+    @GuardedBy("this")
+    private long currentMemoryPoolAssignmentVersion = -1;
```

Comment:
The `QueryStateMachine` starts with `new VersionedMemoryPoolId(GENERAL_POOL, 0)`, so shouldn't we start with zero here also?